### PR TITLE
WINC-650: [wmco] Apply worker role to BYOH Nodes

### PR DESCRIPTION
**Pull request by sebsoto**
The goal of this PR is to ensure that BYOH Nodes have the worker role label.
In order to achieve this changes had to be made to the patch generation functions,
refactoring that makes up the bulk of the commits in the PR.

### CODE REVIEW

**Review by mansikulkarni96 — COMMENTED** [review]
Overall the implementation looks good, I think commit 2 &3 could have been combined together. Changes to the same functions in different commits, seems a little confusing to review.

File: pkg/metadata/metadata.go [hunk 1]
```diff
@@ -12,45 +12,55 @@ import (
...
 const VersionAnnotation = "windowsmachineconfig.openshift.io/version"
 
-// generateAnnotationPatch creates a patch applying the given operation onto each given annotation key and value
-func generateAnnotationPatch(op string, annotations map[string]string) ([]*patch.JSONPatch, error) {
-	if len(annotations) == 0 {
-		return nil, errors.New("annotations to format cannot be empty or nil")
+// generatePatch creates a patch applying the given operation onto each given annotation key and value
+func generatePatch(op string, labels, annotations map[string]string) ([]*patch.JSONPatch, error) {
+	if len(labels) == 0 && len(annotations) == 0 {
+		return nil, errors.New("cannot have both label and annotation maps empty")
```
Comment (aravindhp) [inline pkg/metadata/metadata.go:18]: `labels and annotations empty` will suffice.

**Review by aravindhp — APPROVED** [review]
Thanks for working on this, @sebsoto 

/approve

File: pkg/metadata/metadata.go [hunk 2]
```diff
@@ -12,45 +12,55 @@ import (
...
 	return json.Marshal(patch)
 }
 
-// GenerateRemovePatch creates a comma-separated list of operations to remove all given annotations from an object
-// A "remove" patch fails transactionally if any of the annotations do not exist
-func GenerateRemovePatch(annotations []string) ([]byte, error) {
+// GenerateRemovePatch creates a comma-separated list of operations to remove all given labels and annotations from an
+// object. A "remove" patch fails transactionally if any of the annotations do not exist.
+func GenerateRemovePatch(labels, annotations []string) ([]byte, error) {
+	labelMap := make(map[string]string)
```
[the remark this answers is not in the document]
Comment (sebsoto) [reply pkg/metadata/metadata.go:47]: Not sure what I'd rename it to considering there is already a `labels`

**Comment by mansikulkarni96** [comment]
Thanks for working on this @sebsoto 
/lgtm

**Pull request merged**